docs/MQTT-streamingsource.md

```diff
@@ -0,0 +1,112 @@
+MQTT Streaming Source
+=====================
+
+<a href="https://cdap-users.herokuapp.com/"><img alt="Join CDAP community" src="https://cdap-users.herokuapp.com/badge.svg?t=mqtt"/></a>
+[![Build Status](https://travis-ci.org/hydrator/sftp-actions.svg?branch=develop)](https://travis-ci.org/hydrator/mqtt) [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0) <img src="https://cdap-users.herokuapp.com/assets/cdap-action.svg"/>
+
+This MQTT Streaming Source allows you to subscribe to a specific topic with a MQTT broker as a
+client for messages in a streaming context.
+
+Usage Notes
+-----------
+
+The MQTT Streaming Source will listen to a MQTT broker and subscribe to a given topic. In order to
+do this, we need the host and port of the MQTT broker. If the MQTT protocol is TCP, the port will
+typically be 1883. Otherwise, if the MQTT protocol is SSL, the port will typically be 8889.
+
+When specifying the MQTT Topic, the wildcards '+' and '#' are permitted. The '+' wildcard indicates
+we are selecting all topics on that level of hierarchy. The '#' wildcard indicates that we are
+selecting all topics from that level of hierarchy and children of those topics. This also means that
+if we choose to use it, '#' must be the last character of a topic subscription.
+
+For example, if we have the topic "a/b/c/d", the following will match:
+
+* a/b/c/d
+* a/+/c/d
+* +/+/+/+
+* a/b/c/+
+* \#
+* a/#
+* a/b/c/#
+* +/b/#
+
+While the following will _not_ match:
+
+* a/b/c
+* +/+/+
+* /#
+
+In addition to this, the zero-length topic, "", is permitted. This means that "/a/b" will match
+"+/a/topic" and "/#". Also, "a//b" will match "a/+/b".
+
```

Comment:
Additionally, the zero-length topic, ...
or how about:
```
Additionally, the zero-length topic, "", is permitted:

* "/a/b" will match "+/a/topic" and "/#"
* "a//b" will match "a/+/b"
```
as that might be quicker read.